src/cpp/session/modules/RStudioAPI.cpp

```diff
@@ -42,29 +42,53 @@ module_context::WaitForMethodFunction s_waitForShowDialog;
 ClientEvent showDialogEvent(const std::string& title,
                             const std::string& message,
                             bool prompt,
-                            const std::string& promptDefault)
+                            const std::string& promptDefault,
+                            bool question,
+                            const std::string& ok,
+                            const std::string& cancel,
+                            const std::string& url)
 {
    json::Object data;
    data["title"] = title;
    data["message"] = message;
    data["prompt"] = prompt;
    data["default"] = promptDefault;
+   data["question"] = question;
+   data["ok"] = ok;
+   data["cancel"] = cancel;
+   data["url"] = url;
    return ClientEvent(client_events::kRStudioAPIShowDialog, data);
 }
 
 SEXP rs_showDialog(SEXP titleSEXP,
                    SEXP messageSEXP,
                    SEXP promptSEXP,
-                   SEXP promptDefaultSEXP)
+                   SEXP promptDefaultSEXP,
+                   SEXP questionSEXP,
```

Comment:
Can we make this an enum instead of a bool? I have the feeling that we'll want to support more kinds of dialogs soon (question, info, warning, error, etc).